packages/bento-design-system/src/TextField/createTextField.tsx

```diff
@@ -31,34 +38,62 @@ export function createTextField(config: InputConfig, { Field }: { Field: FieldTy
       inputRef
     );
 
+    const [showPassword, setShowPassword] = useState(false);
+    const passwordIcon = showPassword
+      ? config.passwordHideIcon ?? IconEye
+      : config.passwordShowIcon ?? IconEyeClosed;
+
```

Comment:
The two icons are swapped, it should show `IconEye` when the password is concealed and `IconEyeClosed` when is in clear text